taskvine/src/manager/vine_current_transfers.c

```diff
@@ -119,35 +121,36 @@ void set_throttle(struct vine_manager *m, struct vine_worker_info *w, int is_des
 	}
 }
 
-int vine_current_transfers_set_failure(struct vine_manager *q, char *id)
+int vine_current_transfers_set_failure(struct vine_manager *q, char *id, const char *cachename)
 {
 	struct vine_transfer_pair *p = hash_table_lookup(q->current_transfer_table, id);
+	if (!p) {
+		return 0;
+	}
 
-	int throttled = 0;
-
-	if (!p)
-		return throttled;
-
+	/* if any of the source or to workers don't exist, we shouldn't be surprised by this transfer failure */
```

Comment:
I don't think this comment is right.  As far as I can tell, the elements of the vine_transfer_pair should always be valid, and I don't see any code that sets them to null.  A failed worker causes the transfer record to be removed, not nulled out.